packages/flutter_tools/lib/src/base/version.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+class Version implements Comparable<Version> {
+  static final RegExp versionPattern =
+      new RegExp(r'^(\d+)(\.(\d+)(\.(\d+))?)?');
```

Comment:
I left it off on purpose, so it'd still match something like '1.2.3-foo', in case we ever came across that.